packages/sync/src/modules/Term_Relationships.php

```diff
@@ -126,7 +126,7 @@ public function enqueue_full_sync_actions( $config, $max_items_to_enqueue, $last
 			$items                 = array_chunk( $objects, $term_relationships_full_sync_item_size );
 			$last_object_enqueued  = $this->bulk_enqueue_full_sync_term_relationships( $items, $last_object_enqueued );
 			$items_enqueued_count += count( $items );
-			$limit                 = min( $limit - $objects_count, self::QUERY_LIMIT );
+			$limit                 = min( $max_items_to_enqueue - $items_enqueued_count, self::QUERY_LIMIT );
```

Comment:
'limit' needs to be a number of rows, each item has  `$term_relationships_full_sync_item_size`  rows. we can't use it as the limit for the query, I think.

Reply:
the issue with the existing line of code

`$limit                 = min( $limit - $objects_count, self::QUERY_LIMIT );`

is that $limit is originally set to at 106 by the min statement of
`$limit = min( $max_items_to_enqueue * $term_relationships_full_sync_item_size, self::QUERY_LIMIT );`

So when we reach this point of the code $limit has already been minified to the QUERY_LIMIT so by using $limit in or test its already been truncated to the size of a single query and we only every get 1 execution of the while statement instead of iterating to match the available space in the queue.


Perhaps it would make sense to re-use the original min statement simply removing the current limit from the first value?

`$limit = min( ($max_items_to_enqueue * $term_relationships_full_sync_item_size - $limit), self::QUERY_LIMIT );`

Reply:
scratch that it wouldn't increment in the above as we need to incrementally decrease.

Reply:
`$limit = min( ($max_items_to_enqueue  - $items_enqueued_count) * $term_relationships_full_sync_item_size, self::QUERY_LIMIT );`

So by the above if we take how much we want to enqueue subtract the number of chunks enqueued. and then times it by chunk size that gives us remainder